tests/crescent/hooks/test_hook_order.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from types import MethodType
+from typing import TYPE_CHECKING, Any
+
+from crescent import Bot, Plugin, command, hook
+from crescent.commands.groups import Group
+
+
+async def async_func(*a, **k) -> None:
+    ...
+
+
+class _MockHook:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.__name__ = name
+
+    __repr__ = __str__ = lambda self: self.name
+
+    __class__ = MethodType
+    __func__ = async_func
```

Comment:
what are these used for

Reply:
Makes `inspect.iscoroutinefunction` return True. Think of it as magic ✨

Basically, `inspect.iscoroutinefunction` does this:
```
while isinstance(obj, types.MethodType):
    obj = obj.func

return  # however the decide if a function is coroutine
```

So when it receives MockFunc, it sees the `__class__` is `MethodType` (so `isinstance(obj, MethodType)` returns True), and so it just checks `obj.__func__`, which *is* a coroutine function.

Reply:
Oh and the `__repr__` and `__str__` make debugging easier.